internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -1013,6 +1013,24 @@ func (r *BMACReconciler) newSpokeMachine(bmh *bmh_v1alpha1.BareMetalHost, cluste
 		},
 	}
 	mutateFn := func() error {
+		if machine.ObjectMeta.Annotations == nil {
+			machine.ObjectMeta.Annotations = make(map[string]string)
+		}
+		machine.ObjectMeta.Annotations["metal3.io/BareMetalHost"] = fmt.Sprintf("%s/%s", "openshift-machine-api", bmh.Name)
+		machine.Spec = machinev1beta1.MachineSpec{
+			ProviderSpec: machinev1beta1.ProviderSpec{
+				Value: &runtime.RawExtension{
+					// TODO: replace image checksum and url. Though, the current nonsensical values appear to still work.
+					Raw: []byte(`{
+						"apiVersion": "baremetal.cluster.k8s.io/v1alpha1",
+						"kind": "BareMetalMachineProviderSpec",
+						"image": {
+						"checksum": "http://to.be.determined",
+						"url": "http://to.be.determined"
+						}}`),
```

Comment:
Can we use the checksum and URL values from the master machines created on the spoke cluster. I see same value in the master machine on the spoke cluster and I think we can retrieve and set the same values into the day2 worker node. If not, please suggest.

```
[root@hardprov-fx2-21 ~]# oc get machine -A --kubeconfig=/root/assistedClusterkubeconfig
NAMESPACE               NAME                                   PHASE     TYPE   REGION   ZONE   AGE
openshift-machine-api   assisted-test-cluster-qf7pz-master-0   Running                          42h
openshift-machine-api   assisted-test-cluster-qf7pz-master-1   Running                          42h
openshift-machine-api   assisted-test-cluster-qf7pz-master-2   Running                          42h


root@hardprov-fx2-21 ~]# oc get machine assisted-test-cluster-qf7pz-master-0 -n openshift-machine-api --kubeconfig=/root/assistedClusterkubeconfig -o=jsonpath='{"Checksum: "}{.spec.providerSpec.value.image.checksum} {"\nURL: "}{.spec.providerSpec.value.image.url}{"\n"}'
Checksum: http://192.168.111.35:6181/images/rhcos-48.84.202106091622-0-openstack.x86_64.qcow2/cached-rhcos-48.84.202106091622-0-openstack.x86_64.qcow2.md5sum 
URL: http://192.168.111.35:6181/images/rhcos-48.84.202106091622-0-openstack.x86_64.qcow2/cached-rhcos-48.84.202106091622-0-openstack.x86_64.qcow2


root@hardprov-fx2-21 ~]# oc get machine assisted-test-cluster-qf7pz-master-1 -n openshift-machine-api --kubeconfig=/root/assistedClusterkubeconfig -o=jsonpath='{"Checksum: "}{.spec.providerSpec.value.image.checksum} {"\nURL: "}{.spec.providerSpec.value.image.url}{"\n"}'
Checksum: http://192.168.111.35:6181/images/rhcos-48.84.202106091622-0-openstack.x86_64.qcow2/cached-rhcos-48.84.202106091622-0-openstack.x86_64.qcow2.md5sum 
URL: http://192.168.111.35:6181/images/rhcos-48.84.202106091622-0-openstack.x86_64.qcow2/cached-rhcos-48.84.202106091622-0-openstack.x86_64.qcow2


root@hardprov-fx2-21 ~]# oc get machine assisted-test-cluster-qf7pz-master-2 -n openshift-machine-api --kubeconfig=/root/assistedClusterkubeconfig -o=jsonpath='{"Checksum: "}{.spec.providerSpec.value.image.checksum} {"\nURL: "}{.spec.providerSpec.value.image.url}{"\n"}'
Checksum: http://192.168.111.35:6181/images/rhcos-48.84.202106091622-0-openstack.x86_64.qcow2/cached-rhcos-48.84.202106091622-0-openstack.x86_64.qcow2.md5sum 
URL: http://192.168.111.35:6181/images/rhcos-48.84.202106091622-0-openstack.x86_64.qcow2/cached-rhcos-48.84.202106091622-0-openstack.x86_64.qcow2

```

Reply:
just wonder when those values are added in regular ipi installation?